ethcore/trace/src/types/trace.rs

```diff
@@ -51,6 +101,48 @@ impl CreateResult {
 	}
 }
 
+/// `Create` method. Distinguish between use of `CREATE` and `CREATE2` opcodes in an action.
+#[derive(Debug, Clone, PartialEq)]
+pub enum CreationMethod {
+	/// Create
+	Create,
+	/// Create2
+	Create2,
+}
+
+impl CreationMethod {
+	fn maybe_new(action_type: ActionType) -> Option<Self> {
```

Comment:
Both of the `fn maybe_new` methods look more like they could be a `TryFrom` implementation, but I don't mind the current.

Just checked that it was me suggesting `fn new() -> Option<Self>` :joy: